operations/research-and-development/engineering/infrastructure-engineering/production-readiness-review.md

```diff
@@ -0,0 +1,38 @@
+# Production Readiness Review
+
+The production readiness review (PRR) is a process that identifies the reliability needs of a service, feature (mid to large only) or a significant change to infrastructure. The Mattermost PRR has been inspired from the [production readiness review](https://sre.google/sre-book/evolving-sre-engagement-model/) from the SRE book. A PRR is considered a prerequisite for the Infrastructure team to accept responsibility for managing the production aspects of a service, feature or infrastructure change.
+
+The goals of the readiness review is to improve the followings:
+- System architecture and interservice dependencies
+- Observability (metrics, monitoring, logging)
+- Incident Response
+- Capacity planning
+- Change management
+- Performance, availability, latency and efficiency
+- Security posture
+- Increase awareness and boost collaboration with a goal to built a cloud-native product
+
+This will help to increase the collaboration between Product, Security and Infrastructure teams in order to bridge any gaps about a new service, feature or infrastructure change. The review document is not intended to be constantly updated and it reflects a snapshot of the reality of what is being deployed and the discussions around it. 
+
+## Process
+
```

Comment:
I think we should set a standard promotion window, where the build goes into escrow (for lack of better term) X-days prior to the agreed-upon deployment date.  That X-days can be arbitrary or we can estimate based on how long we feel the validation process will take.  There will also have to be an exception/extended time scenario for XL changes.

Reply:
We could set a minimum of X-days

Reply:
@lieut-data overall I agree, my only worry is to not find any big gaps which can drag the whole implementation back. Of course it's up to us to involve the right people to avoid such cases.

I will agree with @iyampaul and @stafot .